tests/core/validation/test_validation.py

```diff
@@ -0,0 +1,408 @@
+import pytest
+
+from evm.exceptions import (
+    ValidationError,
+)
+from evm.constants import (
+    SECPK1_N,
+)
+from evm.validation import (
+    validate_block_number,
+    validate_canonical_address,
+    validate_gt,
+    validate_gte,
+    validate_is_boolean,
+    validate_is_bytes,
+    validate_is_integer,
+    validate_length,
+    validate_length_lte,
+    validate_lt,
+    validate_lte,
+    validate_lt_secpk1n,
+    validate_lt_secpk1n2,
+    validate_multiple_of,
+    validate_multiple_of_8,
+    validate_stack_item,
+    validate_uint256,
+    validate_unique,
+    validate_vm_block_numbers,
+    validate_word,
+)
+
+
+@pytest.mark.parametrize(
+    "value,is_valid",
+    (
+        (tuple(), True),
+        ([], True),
+        ({}, True),
+        (set(), True),
+        ([1, 2, 3], True),
+        ([1, 2, 1], False),
+        (['a', 'b', 'c'], True),
+        (['1', '2', '3'], True),
+        (['1', '2', '1'], False),
+        (['1', '2', 1], True),
+    ),
+)
+def test_validate_unique(value, is_valid):
+    if is_valid:
+        validate_unique(value)
+    else:
+        with pytest.raises(ValidationError):
+            validate_unique(value)
+
+
+@pytest.mark.parametrize(
+    "value,is_valid",
+    (
+        (tuple(), False),
+        ([], False),
+        ({}, False),
+        (set(), False),
+        (1, False),
+        (True, False),
+        ('abc', False),
+        (b'abc', True),
+    ),
+)
+def test_validate_is_bytes(value, is_valid):
+    if is_valid:
+        validate_is_bytes(value)
+    else:
+        with pytest.raises(ValidationError):
+            validate_is_bytes(value)
+
+
+@pytest.mark.parametrize(
+    "value,is_valid",
+    (
+        (tuple(), False),
+        ([], False),
+        ({}, False),
+        (set(), False),
+        (1234, True),
+        ('abc', False),
+        # (True, False),
```

Comment:
If this is failing then the validation function should be fixed.  I suspect that the `is_integer` function from `eth_utils` is returning `True` for `is_integer(True)` which is a bug.  If this is the case, can you open up an issue in the [`ethereum-utils`](https://github.com/pipermerriam/ethereum-utils) repository.